vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java

```diff
@@ -484,7 +484,7 @@ public void setItemLabelGenerator(
         Objects.requireNonNull(itemLabelGenerator,
                 "The item label generator can not be null");
         this.itemLabelGenerator = itemLabelGenerator;
-        reset();
+        refreshCheckboxes();
```

Comment:
Should we align the implementation with [that of RadioButtonGroup](https://github.com/vaadin/flow-components/blob/b5a97698b1fc895bdb64d06c3e6b41abd8c38d0c/vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java#L366)?

Reply:
Good point. The implementation in RBG seems to make more sense, but we should have a discussion about this in the team.

Ideally the JavaDocs should be improved as well, to mention that setting an item label generator overrides / removes the renderer.

Reply:
I will create a separate issue to align the behaviour of the components.